### PR TITLE
[TCFMM-26] Entrypoint for getting position info

### PR DESCRIPTION
**Pull request by Martoon-00**
[//]: # (This is a template of a pull request template.)
[//]: # (You should modify it considering specifics of a particular repository and put it there.)
[//]: # (Comments like this are meta-comments, they shouldn't be present in the final template.)
[//]: # (Comments starting with '

[//]: # (Here you can add a link to the corresponding issue tracker, e. g. https://issues.serokell.io/issue/AD-)
[//]: # (For GitHub/GitLab issues it is better to use just hash symbol or exclamation mark as it is more resistant to repo movements)
[//]: # (In this case please also prefix the link with "Resolves" keyword)
[//]: # (See https://docs.gitlab.com/ee/user/project/issues/managing_issues.html#closing-issues-automatically)
[//]: # (See https://help.github.com/en/github/managing-your-work-on-github/closing-issues-using-keywords)
## Related issue(s)

Helps for TCFMM-26

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [x] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

[//]: # (Add more docs here if you have them in the repository)
- Documentation
  - [x] I checked whether I should update the docs and did so if necessary:
    - [Specification](../tree/master/docs/specification.md)
    - [README](../tree/master/README.md)
    - Haddock

#### Stylistic guide (mandatory)

[//]: # (Update link to style guide if necesary or remove if it's not present)

- [x] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [x] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: ligo/types.mligo [hunk 1]
```diff
@@ -372,7 +372,9 @@ type y_to_x_rec_param = x_to_y_rec_param
 
 type oracle_view_param = cumulatives_value list
 
-type observe_param = {
+type observe_param =
+[@layout:comb]
```
Comment (pasqu4le) [inline ligo/types.mligo:376]: Not that I'm particularly against it, but why the change in layout?

File: ligo/main.mligo [hunk 2]
```diff
@@ -365,6 +365,19 @@ let update_timed_cumulatives (s : storage) : storage =
         }
         in {s with cumulatives_buffer = new_buffer}
 
+let get_position_info (s : storage) (p : get_position_info_param) : result =
+    let position_index = match Big_map.find_opt p.position_id s.position_indexes with
+        | None -> (failwith position_not_exist_err : position_index)
+        | Some v -> v in
```
Comment (pasqu4le) [inline ligo/main.mligo:371]: It looks like we have `get_position_index` for this, it returns an FA2-specific error, but IMO that's still valid here.

File: ligo/main.mligo [hunk 3]
```diff
@@ -365,6 +365,19 @@ let update_timed_cumulatives (s : storage) : storage =
...
+    let position = match Big_map.find_opt position_index s.positions with
+        | None -> (failwith internal_position_maps_unsynced_err : position_state)
+        | Some v -> v in
+    let result = {
+        liquidity = position.liquidity;
+        index = position_index;
+    } in
```
Comment (pasqu4le) [inline ligo/main.mligo:378]: Could be reduced to:
```suggestion
    let result = match Big_map.find_opt position_index s.positions with
        | None -> (failwith internal_position_maps_unsynced_err : position_state)
        | Some position ->
          { liquidity = position.liquidity
          ; index = position_index;
          }
    in
```

File: ligo/errors.mligo [hunk 4]
```diff
@@ -137,6 +140,12 @@
 (* Some issue with binary search in `observe` entrypoint. *)
 [@inline] let internal_observe_bin_search_failed = 322n
 
+(* Attempt to garbade collect a tick with non-zero liquidity net. *)
+[@inline] let internal_non_empty_position_gc_err = 323n
```
Comment (pasqu4le) [inline ligo/errors.mligo:144]: It looks like this isn't used anywhere :thinking:

File: docs/specification.md [hunk 5]
```diff
@@ -459,6 +460,38 @@ type set_position_param = {
...
+
+### **get_position_info**
+
+Obtains information about the position with the given identifier.
+The used identifier is `token_id` that is used to select the position in `transfer` / `get_balance` FA2 entrypoints.
+
+- `liquidity` is the current amount of liquidity added by this position.
+- `lower_tick_index` and `upper_tick_index` are the boundary ticks in which the position is active. These values are constant for the given position.
+- `owner` is the current position's owner in accordance to the position's semantics of FA2 asset.
+-- If the specified position identifier does not exist, fails with `position_not_exist_err` error code.
```
Comment (pasqu4le) [inline docs/specification.md:472]:
```suggestion
- If the specified position identifier does not exist, fails with `position_not_exist_err` error code.
```

File: docs/specification.md [hunk 6]
```diff
@@ -459,6 +460,38 @@ type set_position_param = {
 type balance_nat = {x : nat ; y : nat}
 ```
 
+
+### **get_position_info**
```
Comment (pasqu4le) [inline docs/specification.md:464]: It would be nice to mention this entrypoint in the explanation in the Liquidity Mining section IMO

File: ligo/types.mligo (hunk 1 above)
Comment (Martoon-00) [reply ligo/types.mligo:376]: Without it, LIGO will reorder fields, and this will not comply with the TZIP View pattern. I supposed it is better to comply with it rather than not )

File: ligo/main.mligo (hunk 2 above)
Comment (Martoon-00) [reply ligo/main.mligo:371]: Oh nice observation, I missed the existence of `get_position_index`.

In the solidity implementation of liquidity mining contract they treat positions very much like tokens, even call the respective variable `token_id`, not `position_id`, so I agree that having FA2-specific error can be quite justified here.

File: ligo/errors.mligo (hunk 4 above)
Comment (Martoon-00) [reply ligo/errors.mligo:144]: This is an artifact, someone (me?) forgot to regenerate errors after updating `generate_error_codes.hs`.

File: docs/specification.md (hunk 5 above)
Comment (Martoon-00) [reply docs/specification.md:472]: Oh, got bad habits after working with Notion :sweat_smile:

File: ligo/errors.mligo (hunk 4 above)
Comment (pasqu4le) [reply ligo/errors.mligo]: Oh I see, got it :ok_hand:

**Review by pasqu4le — APPROVED** [review]
LGTM now
